src/core_modules/capture-core/rules/actionsCreator/runRulesForEnrollmentPage.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+import { RulesEngine } from '../engine';
+import type { Program } from '../../metaData';
+import type {
+    EventValues,
+    OrgUnit,
+    DataElement,
+    TEIValues,
+    Enrollment,
+    TrackedEntityAttributes,
+} from '../engine';
+import { constantsStore } from '../../metaDataMemoryStores/constants/constants.store';
+import { optionSetStore } from '../../metaDataMemoryStores/optionSets/optionSets.store';
+
+
+export default function runRulesForEnrollmentPage(
+    program: Program,
+    orgUnit: OrgUnit,
+    allEventsData: EventValues,
+    dataElementsInProgram: { [string]: DataElement },
+    teiValues: ?TEIValues,
+    trackedEntityAttributes: ?TrackedEntityAttributes,
+    enrollmentData: ?Enrollment,
+) {
+    const { programRuleVariables: programRulesVariables } = program;
+    const programRules = [...program.programRules];
+
+    const constants = constantsStore.get();
+    const optionSets = optionSetStore.get();
+
+    // returns an array of effects that need to take place in the UI.
+    return RulesEngine.programRuleEffectsForEnrollment(
+        { programRulesVariables, programRules, constants },
+        allEventsData,
+        orgUnit,
+        dataElementsInProgram,
+        teiValues,
+        trackedEntityAttributes,
+        enrollmentData,
+        // $FlowFixMe[prop-missing] automated comment
```

Comment:
The `optionSets: OptionSets` is set bellow in `static programRuleEffectsForEnrollment`, I wonder then why the `$FlowFixMe` comment is needed. Can you help me understand? Thanks!

Reply:
@simonadomnisoru This is something we should add to our Flow vs Typescript comparison! The input object is a type that contains all the properties the param expects, but it also contains some additional properties. I would expect this to work when the type defined for the param is inexact (meaning more properties than those declared should be accepted). Tried to do some magic with the $Shape utility type, but also ran into problems with that approach.

I would say keep the $FlowFixMe for now, but maybe @huocha can add the label TypescriptCompare (or something similar) so that we remember to check how this is working with Typescript when the time comes.